lib/livebook_web/live/session_live.ex

```diff
@@ -1223,6 +1266,29 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, socket}
   end
 
+  def handle_info({:set_secret, secret}, socket) do
+    idx = Enum.find_index(socket.assigns.livebook_secrets, &(&1.name == secret.name))
```

Comment:
I wonder if we should keep them as a map and then sort in the view just before rendering?